src/action/install_date.rs

```diff
@@ -0,0 +1,284 @@
+            // Not a device file name.
+            return None;
+        }
+        Some(device)
+    }
+
+    /// Parses `dumpe2fs` output to get the filesystem creation time.
+    ///
+    /// This function returns `None` in case of errors.
+    fn parse_creation_time_from_dumpe2fs<R>(output: R) -> Option<SystemTime>
+    where
+        R: Read,
+    {
+        let reader = BufReader::new(output);
+        for line in reader.lines() {
+            const FIELD_NAME: &'static str = "Filesystem created:";
+            let line = line.ok()?;
+            if !line.starts_with(FIELD_NAME) {
+                continue;
+            }
+            let line = line[FIELD_NAME.len() + 1 ..].trim();
+            let local_time = Local.datetime_from_str(line, "%c").ok()?;
+            let utc_time = local_time.with_timezone(&Utc);
+            let time_since_epoch =
+                (utc_time - Utc.timestamp(0, 0)).to_std().ok()?;
+            return Some(time::UNIX_EPOCH + time_since_epoch);
+        }
+        None
+    }
+
+    /// Launches `dumpe2fs`, captures its output and obtains filesystem
+    /// creation time.
+    ///
+    /// This function returns `None` in case of errors.
+    pub fn creation_time_from_dumpe2fs() -> Option<SystemTime> {
+        let output = Command::new("dumpe2fs")
+            .arg(get_root_device()?)
+            .stdout(Stdio::piped())
+            .spawn().ok()?
+            .wait_with_output().ok()?;
+        if !output.status.success() {
+            return None;
+        }
+        parse_creation_time_from_dumpe2fs(Cursor::new(output.stdout))
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+        use std::io::Cursor;
+
+        #[test]
+        fn test_parse_creation_time() {
+            let correct = DateTime::parse_from_rfc3339("2020-05-31T19:02:03Z")
+                .unwrap().with_timezone(&Utc);
+            let creation_time = correct.with_timezone(&Local)
+                .format("%c").to_string();
+            let input = String::from(
+                "Line 1\n\
+                 Line 2\n\
+                 Filesystem created:    "
+            ) + &creation_time + "\n";
+            let cursor = Cursor::new(input.as_bytes());
+            let parsed = parse_creation_time_from_dumpe2fs(cursor).unwrap();
+            let parsed = DateTime::<Utc>::from(parsed);
+            assert_eq!(parsed, correct);
+        }
+    }
+}
+
+/// Obtains system install date (Linux version).
+///
+/// To obtain install date, it first checks the creation time of the root
+/// filesystem. Then, it searches for the installation logs and gets their
+/// modification date. It also checks for other files that were probably
+/// unchanged from the installation.
+///
+/// This function returns `None` in case of errors.
+#[cfg(target_os = "linux")]
+fn get_install_time() -> Option<SystemTime> {
+    // First, check the creation time of the root. This method works on
+    // Linux >= 4.11.
```

Comment:
Could you make this comment more detailed (e.g. why doesn't it work on Linux pre-4.11)?